### PR TITLE
[ci/release] Increase long running timeout, fix artifacts copy

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

With the new job-based file copy, fetching results takes longer. We thus have to increase the long running update test check times in order not to run into bogus release test failures.
Also fixes artifact uploading issues.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Looks great, thanks Kai!  Looks like this should close https://github.com/ray-project/ray/issues/21431.

File: release/run_e2e.sh [hunk 1]
```diff
@@ -98,6 +98,8 @@ pip install -q -r requirements.txt
 pip install -q -U boto3 botocore
 git clone -b "${RAY_TEST_BRANCH}" "${RAY_TEST_REPO}" ~/ray
 
+rm -rf "${RELEASE_RESULTS_DIR-/tmp/ray_release_test_artifacts/}" || true
```
Comment (mwtian) [inline release/run_e2e.sh:101]: Should this use `sudo`, and run inside the while loop?

Comment (krfricke) [reply release/run_e2e.sh]: Ah good call, yep, moved it

**Review by mwtian — APPROVED** [review]
Great!

**Comment by krfricke** [comment]
Test run here: https://buildkite.com/ray-project/periodic-ci/builds/2528

**Comment by mwtian** [comment]
Looks like the output directory still exists: https://buildkite.com/ray-project/periodic-ci/builds/2528#1d3917b2-ba12-423b-b6f7-3cc71229b2ab/113-397. Maybe we need to `sudo rm -rf` in e2e.py just before copying? Also I don't understand how this works: https://github.com/ray-project/ray/blob/master/release/.buildkite/build_pipeline.py#L311-L321, but can it be related?

**Comment by krfricke** [comment]
I think the changes weren't picked up in the run (I removed the `cp` command) - I have to push this as a branch. Will do that now...

**Comment by krfricke** [comment]
Build: https://buildkite.com/ray-project/periodic-ci/builds/2529

**Comment by krfricke** [comment]
Confirmed this uploads artifacts correctly: https://buildkite.com/ray-project/periodic-ci/builds/2529#46e2108a-e04d-4409-ae49-19ff85aba833

I'll merge this PR

**Comment by mwtian** [comment]
Nice fix!

**Pull request merged**